llmfoundry/models/hf/hf_base.py

```diff
@@ -0,0 +1,449 @@
+
+log = logging.getLogger(__name__)
+
+
+class BaseHuggingFaceModel(HuggingFaceModel):
+    """Wrapper around HuggingFaceModel.
+
+    Base class for HuggingFace based models.
+    """
+
+    model_cls: Union[_BaseAutoModelClass,
+                     PreTrainedModel] = AutoModelForCausalLM
+    default_train_metrics: tuple = ()
+    default_eval_metrics: tuple = ()
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path: str,
+        tokenizer: Optional[PreTrainedTokenizerBase] = None,
+        pretrained: bool = True,
+        pretrained_lora_id_or_path: Optional[str] = None,
+        trust_remote_code: bool = True,
+        use_auth_token: bool = False,
+        use_flash_attention_2: bool = False,
+        load_in_8bit: bool = False,
+        init_device: str = 'cpu',
+        config_overrides: Optional[dict[str, Any]] = None,
+        use_logits: bool = True,
+        shift_labels: bool = False,
+        peft_config: Optional['PeftConfig'] = None,
+        allow_embedding_resizing: bool = False,
+        use_train_metrics: bool = True,
+        additional_train_metrics: Optional[list] = None,
+        additional_eval_metrics: Optional[list] = None,
+        should_save_peft_only: bool = True,
+    ):
+        config_overrides = config_overrides or {}
+
+        model = self.build_inner_model(
+            pretrained_model_name_or_path=pretrained_model_name_or_path,
+            pretrained_lora_id_or_path=pretrained_lora_id_or_path,
+            trust_remote_code=trust_remote_code,
+            init_device=init_device,
+            use_flash_attention_2=use_flash_attention_2,
+            use_auth_token=use_auth_token,
+            config_overrides=config_overrides,
+            load_in_8bit=load_in_8bit,
+            pretrained=pretrained,
+        )
+
+        model = self.transform_model(model)
+
+        self.prepare_inner_model(model, init_device)
+
+        metrics, eval_metrics = self.build_metrics(
+            use_train_metrics=use_train_metrics,
+            additional_train_metrics=additional_train_metrics,
+            additional_eval_metrics=additional_eval_metrics,
+        )
+
+        if peft_config is not None and not peft_installed:
+            raise ValueError(
+                'PEFT is not installed, but peft_config was passed. Please install LLM Foundry with the peft extra to use peft_config.',
+            )
+
+        peft_config_object = None
+        if peft_config is not None:
+            peft_config_object = self.get_peft_config(peft_config)
+
+        super().__init__(
+            model=model,
+            tokenizer=tokenizer,
+            use_logits=use_logits,
+            metrics=metrics,
+            eval_metrics=eval_metrics,
+            shift_labels=shift_labels,
+            allow_embedding_resizing=allow_embedding_resizing,
+            peft_config=peft_config_object,
+            should_save_peft_only=should_save_peft_only,
+        )
+
+    def forward(self, batch: Mapping):
```

Comment:
This is essentially identical to the forward in the base class (https://github.com/mosaicml/composer/blob/72b2507b8dd63b49891d099c0540740571c100f1/composer/models/huggingface.py#L484-L493), so lets just remove while we're here.